src/modules/hip/kernel/slice.hpp

```diff
@@ -2,24 +2,79 @@
 #include <omp.h>
 #include "rpp_hip_common.hpp"
 
+template <typename T>
+__global__ void fill_value_ncdhw_hip_tensor(T *dstPtr,
+                                            uint3 dstStridesCDH,
+                                            int channels,
+                                            uint3 dstDimsDHW,
+                                            T *fillValue)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;        // W - inner most dim vectorized
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;              // H - second to inner
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;              // D - outer most dim
+
+    if ((id_z >= dstDimsDHW.x) || (id_y >= dstDimsDHW.y) || (id_x >= dstDimsDHW.z))
+    {
+        return;
+    }
+
+    uint dstIdx = (id_z * dstStridesCDH.y) + (id_y * dstStridesCDH.z) + id_x;
+    d_float8 val_f8;
+    val_f8.f4[0] = (float4)(*fillValue);
```

Comment:
Instead of checking if fillValue is needed, and running a separate kernel like this, have we evaluated how the performance is on simply doing a hipMemset for the whole buffer?

Reply:
hipMemset, memset only works well when the value we want to memset is 0
For other values, it is recommended. Below is one online reference i found for the same
https://stackoverflow.com/questions/51053705/initialize-float-type-array-using-memset